# Free App Profiles With the Most Users

The goal of this project is to determine what type of free apps are likely to attract more users. This information will be used for the purpose of informing developers which apps to place in-add ads to maximize revenue.

In this hypothetical scenario, I will be working as a data analyst for a company that builds Android and iOS mobile apps, and my job is to enable our team of developers to make data-driven decisions with respect to the kind of apps they build. The aim is to find mobile app profiles that are profitable for the App Store and Google Play markets.

At my company, we only build apps that are free to download and install, and our main source of revenue consists of in-app ads. This means that our revenue for any given app is mostly influenced by the number of users that use our app. Our goal for this project is to analyze data to help our developers understand what kinds of apps are likely to attract more users.

## The Data Sets

As of September 2018, there were approximately 2 million iOS apps available on the App Store, and 2.1 million Android apps on Google Play.

I will be looking at a sub-section of all of the apps in the following existing data sets:

A [data set](https://www.kaggle.com/lava18/google-play-store-apps) containing data about approximately 10,000 Android apps from Google Play; the data was collected in August 2018. You can download the data set directly from [this link](https://dq-content.s3.amazonaws.com/350/googleplaystore.csv).

A [data set](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps) containing data about approximately 7,000 iOS apps from the App Store; the data was collected in July 2017. You can download the data set directly from [this link](https://dq-content.s3.amazonaws.com/350/AppleStore.csv).

## Exploring the Data Sets

We'll begin by opening the data sets and extracting their headers.

In [1]:
from csv import reader

# Googe Play data set
opened_file = open('googleplaystore.csv')
read_file = reader(opened_file)
android = list(read_file)
android_header = android[0]
android_data = android[1:]

# Apple Store data set
opened_file = open('AppleStore.csv')
read_file = reader(opened_file)
ios = list(read_file)
ios_header = ios[0]
ios_data = ios[1:]

We will then explore the data by identifying columns that could help with our analysis.

Let's do this by printing the first few rows of each data set and finding the number of rows and columns for each. We will create a function to do this since we may want to explore different sections of the data.

In [2]:
def explore_data(dataset, start, end, rows_and_columns = False):
    piece_of_dataset = dataset[start:end]
    for row in piece_of_dataset:
        print(row)
        print('\n')
    
    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))

For Android:

In [3]:
print(android_header)
print('\n')
explore_data(android_data, 0, 5, True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Eve

The Google Play data set has 10841 apps each with 13 columns or attributes. Attributes such as `'App'`, `'Category'`, `'Rating'`, `'Reviews'`, `'Installs'`, `'Type'`,  `'Price'`, and `'Genres'` may be useful for our analysis.

For iOS:

In [4]:
print(ios_header)
print('\n')
explore_data(ios_data, 0, 5, True)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


['420009108', 'Temple Run', '65921024', 'USD', '0.0', '1724546', '3842', '4.5', '4.0', '1.6.2', '9+', 'Games', '40', '5', '1', '1']


['284035177', 'Pandora - Music & Radio', '130242560', 'USD', '0.0', '1126879', '3594', '4.0', '4.5', '8.4.1', '12+', 'Music', '37', '4', '1', '1']


Number of rows: 7197
Number of colums: 16


The Apple Store data set has 7197 apps each with 16 columns or attributes. Attributes such as `'track_name'`, `'price'`, `'rating_count_tot'`,`'rating_count_ver'`, `'user_rating'`, and `'prime_genre'` may be useful for our analysis.

The data set columns may not be self-explanatory so documentation can be found [here](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps/home).

# Cleaning the Data

Before beginning our analysis, we need to ensure that there are no inaccurate data entries. We need to: 

* Detect inaccurate data, and correct or remove it.
* Detect duplicate data, and remove duplicates.

We are also working for a company that only builds *free* apps and are directed toward an *English speaking audience*. We need to:

* Remove non-English apps
* Remove apps that aren't free

## Cleaning the Google Play data set
The Google Play data set has a dedicated [discussion section]https://www.kaggle.com/lava18/google-play-store-apps/discussion that outlines an error for entry 10472. Let's see what that entry looks like and compare it against correct rows.

In [5]:
print(android_data[10472]) #incorrect row
print('\n')
print(android_header)
print('\n')
print(android_data[0]) #correct row

['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


The incorrect row only has 12 columns, inclusive of the blank `''` entry. We see that the row is missing the `'Genre'` entry. We also notice the `'Category'` column is missing - it is innocrectly classified as `'1.9'`. This is indicative that a deletion of the `'Category'` entry has shifted the other attributes one cell to the left.

It is easiest if we do not include this row in our analysis. Let's delete it.

In [6]:
print(len(android_data))
del android_data[10472]
print(len(android_data))

10841
10840


## Cleaning the iOS data set

On inpsection of the App Store data set [discussion section](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps/discussion), one way we can find errors is to search if each row does not have 16 entries or columns. Let's go through each row and count the length of entries, and print the ones that are not equal to 16.

In [7]:
for row in ios_data:
    if len(row) !=len(ios_header):
        print(row) #prints the row data
        print('\n')
        print(ios_data.index(row)) #prints the row no.

Each row has the expected number of entries. Great! It appears that no information is missing. Let's move on for now.

## Deleting Duplicate Data - Google Play

If we explore the Google Play data set long enough or look at the [discussions section](https://www.kaggle.com/lava18/google-play-store-apps/discussion), you'll notice some apps have duplicate entries. For instance, Instagram has four entries:

In [8]:
for app in android:
    name = app [0]
    if name == 'Instagram':
        print(app)

['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


Let's check if there are more duplicates and locate where they are. We will store the duplicates' names in a separate list called `duplicate_apps` and isolate the unique apps in its own list called `unique_apps`.

In [9]:
duplicate_apps = []
unique_apps = []

for row in android_data:
    name = row[0]
    if name in unique_apps:
        duplicate_apps.append(name)
    else:
        unique_apps.append(name)

print('Number of duplicate apps:', len(duplicate_apps))
print('\n')
print('Examples of duplicate apps:', duplicate_apps[:15])

Number of duplicate apps: 1181


Examples of duplicate apps: ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack', 'FreshBooks Classic', 'Insightly CRM', 'QuickBooks Accounting: Invoicing & Expenses', 'HipChat - Chat Built for Teams', 'Xero Accounting Software']


We can see that the Google Play data set has 1181 repeated apps, but many are repeated more than once. For instance, from the list printed above, we see that the app `'Box'` is repeated. 

We won't remove the duplicates randomly in the `duplicate_apps` list. We'll choose the duplicate apps with the highest number of reviews. We can see from the `'Instagram'` apps printed above, that they have varying number of reviews. We'll isolate the entry with the highest number of reviews, plausibly indicating the most recent one, and delete the rest.

We will do this by creating a dictionary, where each dictionary key is a unique app name and the corresponding dictionary value is the highest number of reviews of that app. We will then use that information stored in the dictionary and create a new data set, which will now only have one entry per app (the one with the highest number of reviews).

We will use the `not in` operator to check for non-membership in a dictionary.

In [10]:
# Creation of dictionary where each key is a unique app name 
# and the corresponding dictionary value is the highest number of reviews
reviews_max = {}
for row in android_data:
    name = row[0]
    n_reviews = float(row[3])
    
    if name in reviews_max and reviews_max[name] < n_reviews:
        reviews_max[name] = n_reviews
        
    elif name not in reviews_max:
        reviews_max[name] = n_reviews

The length of the duplicates list is 1,181, so we should expect that our dictionary `reviews_max` is the difference between our data set and 1,181.

In [11]:
print('Length Data Set:', len(android_data))
print('Length Dictionary:', len(reviews_max))
print('Difference:', len(android_data) - len(reviews_max))

Length Data Set: 10840
Length Dictionary: 9659
Difference: 1181


We see that the difference is 1181 - what we expected.

Now let's use the dictionary above to remove duplicate rows. We loop through the android data set, and for every iteration:

* We isolate the name of the app and the number of reviews.
* We add the current row (app) to the android_clean list, and the app name (name) to the already_added list if:
    * The number of reviews of the current app matches the number of reviews of that app as described in the reviews_max dictionary; and
    * The name of the app is not already in the already_added list. We need to add this supplementary condition to account for those cases where the highest number of reviews of a duplicate app is the same for more than one entry (for example, the Box app has three entries, and the number of reviews is the same). If we just check for reviews_max[name] == n_reviews, we'll still end up with duplicate entries for some apps.

In [12]:
# Create two empty lists to store clean data and another for app names
android_clean = []
already_added = []

for row in android_data:
    name = row[0]
    n_reviews = float(row[3])
    
    if (reviews_max[name] == n_reviews) and (name not in already_added):
        android_clean.append(row)
        already_added.append(name) # keeps track of apps already added

#Inspect some entries to see if our output is what we expected
print(android_clean[9])
print('\n')
print(already_added[1077:1080])
print('\n')

explore_data(android_clean, 1077, 1080, True)

len(already_added)

['Text on Photo - Fonteee', 'ART_AND_DESIGN', '4.4', '13880', '28M', '1,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'October 27, 2017', '1.0.4', '4.1 and up']


['Realtor.com Real Estate: Homes for Sale and Rent', 'Real Estate sale & rent Trovit', '591 housing transactions - renting houses, middle-class houses, new cases, real-time registration, villas through the sky, apartment suites, MRT, buying a house selling prices, housing mortgages']


['Realtor.com Real Estate: Homes for Sale and Rent', 'HOUSE_AND_HOME', '4.5', '162243', '12M', '10,000,000+', 'Free', '0', 'Everyone', 'House & Home', 'July 26, 2018', '8.18', '4.0.3 and up']


['Real Estate sale & rent Trovit', 'HOUSE_AND_HOME', '4.1', '65913', '7.7M', '5,000,000+', 'Free', '0', 'Everyone', 'House & Home', 'July 16, 2018', '4.47.3', '4.0.3 and up']


['591 housing transactions - renting houses, middle-class houses, new cases, real-time registration, villas through the sky, apartment suites, MRT, buying a house selling pr

9659

The entries' formats look correct. And we have the expected 9,659 rows in the `android_clean` data set. Excellent!

## Removal of Non-English Apps
Remember our company only develops apps in English. There are, however, apps in the data set that are non-English, which we would like to remove.

One way to do this is to remove each app with a name containing a symbol that is not commonly used in English text. Behind the scenes, each character we use in a string has a corresponding number associated with it. We can retreive the corresponding number of each character by using the [built-in function](https://docs.python.org/3/library/functions.html#ord) `ord()`. For instance:

In [13]:
print(ord('a'))
print(ord('A'))
print(ord('1'))
print(ord('!'))

97
65
49
33


The numbers corresponding to the characters we commonly use in the English language are all in the range of 0 to 127 according to ASCII (American Standard Code of Informaiton Interchange). Based on this number range, we can build a function that detects whether a character belongs to the set of common English characters or not. If the number is equal to or less than 127, the the character belongs to the set of common English characters.

Let's index through each of the app's name by writing a function that does the takes in a string and returns `False` if there are any characters in the string that does not belong to ASCII 0 - 127, otherwise it returns `True`:

* Iterates over the input string and checks whether each character is associated with the a character greater than 127. If so, the function should `return False` and the app name is probably non-English.
* If the loop finishes running without the `return` statement being executed, then it means no character had a corresponding number over 127 and the app name is probably English, so the function returns `True`.

In [14]:
def check_if_English(string):
 
    for character in string:
        if ord(character) > 127:
            return False
        
    return True
    
# Check efficacy of code
print(check_if_English('Instagram'))
print(check_if_English('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print(check_if_English('Docs To Go™ Free Office Suite'))
print(check_if_English('Instachat 😜'))

True
False
False
False


We see that some symbols like `™` or `😜` `return False` even though they are English apps. Let's dissect this a bit more:

In [15]:
print(ord('™'))
print(ord('😜'))

8482
128540


These symbols are > 127. How can we optimize this? While not perfect, and to prevent spending too much time on this, let's create an algorithm that says `if` there are more than three non-ASCII character greater than 127, we will label it as non-English.

In [16]:
def check_if_English(string):
    
    non_ascii = 0  

    for character in string:
        if ord(character) > 127:
            non_ascii += 1
            
    if non_ascii > 3:
        return False
    else:     
        return True

# Check efficacy of code
print(check_if_English('Instagram'))
print(check_if_English('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print(check_if_English('Docs To Go™ Free Office Suite'))
print(check_if_English('Instachat 😜'))

True
False
True
True


Now let's use the `check_if_english` function to filter out non-English apps from both data sets. We will loop through each data set and, if the app is identified as English, we will append the whole row to a separate list.

In [17]:
android_english_apps = []
ios_english_apps = []


for row in android_clean:
    name = row[0]
    if check_if_English(name):
        android_english_apps.append(row)

for row in ios_data:
    name = row[1]
    if check_if_English(name):
        ios_english_apps.append(row)
    
explore_data(android_english_apps, 6, 8, True)
print('\n')
explore_data(ios_english_apps, 6, 8, True)
    

['Infinite Painter', 'ART_AND_DESIGN', '4.1', '36815', '29M', '1,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'June 14, 2018', '6.1.61.1', '4.2 and up']


['Garden Coloring Book', 'ART_AND_DESIGN', '4.4', '13791', '33M', '1,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'September 20, 2017', '2.9.2', '3.0 and up']


Number of rows: 9614
Number of colums: 13


['282935706', 'Bible', '92774400', 'USD', '0.0', '985920', '5320', '4.5', '5.0', '7.5.1', '4+', 'Reference', '37', '5', '45', '1']


['553834731', 'Candy Crush Saga', '222846976', 'USD', '0.0', '961794', '2453', '4.5', '4.5', '1.101.0', '4+', 'Games', '43', '5', '24', '1']


Number of rows: 6183
Number of colums: 16


We are left with 9614 Android apps and 6183 iOS apps.

# Isolating Free Apps

So far, we have cleaned the data by:

Removing inaccurate data
Removing duplicate app entries
Removing non-English apps

Our company only builds free-to-download apps and our main source of revenue consists of in-app ads. So far, our data consists of `free` and `non_free` apps. Isolating the `free` apps will be the last step in cleaning the data.

In [18]:
#Loop through each data set to isolate the free apps in a separate list
android_free = []
ios_free = []

for row in android_english_apps:
    price = row[7]
    if price == '0':
        android_free.append(row)
        
for row in ios_english_apps:
    price = float(row[4])
    if price == 0:
        ios_free.append(row)

#Check the length of each data set to see how many apps we have remaining
explore_data(android_free, 0, 3, True)
print('\n')
explore_data(ios_free, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 8864
Number of colums: 13


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', '

We have 8864 Android apps and 3222 iOS apps, which should be enough for our analysis.

# Frequency Tables to Determine the Most Common Genres in Google Play and the App Store

Let's re-state our goal. Our aim is to determine the kinds of apps that are likely to attract the most users. More users correlated to higher revenue since these are the apps where we would have ads.

To minimize the risk and overhead, our validation strategy for an app idea is comprised of the following:

1. Build a MVP, or minimal viable product. In this case, this is the Android version of the app, and add it to Google Play.
2. If the app has a good response from users, we develop it further.
3. If the app is profitable after 6 months, we build an iOS version of the app and add it to the App Store.

Our end goal is to have the app successful on both Google Play and the App Store. This means that we will need to find the types of apps that do well on both markets. 

To begin to determine what types of apps are most successful, let's get a feel for what the most common genres are for each market.

For the android apps, we will look at the columns `Category` and `Genre` in the Google Play data set. For iOS apps, we will look at the column `prime_genre` in the App Store data set.

We will build two functions. One to create a frequency table and its percentages, with two inputs: the data set we want to parse through, and the column number of  the data set.

The other will display the percentages in descending order.

In [19]:
def freq_table(data_set, index):
    table = {}
    total = 0
    for row in data_set:
        total +=1
        value = row[index]
        if value in table:
            table[value] +=1
        else:
            table[value] = 1
    
    table_percentages = {}
    for key in table:
        percentage = (table[key] / total) * 100
        table_percentages[key] = percentage

    return table_percentages

# category_android = freq_table(android_free, 1)
# genre_android = freq_table(android_free, 9)
# genre_ios = freq_table(ios_free, 11)

# print(category_android)
# print('\n')
# print(genre_android)
# print('\n')
# print(genre_ios)

Great! It looks like we have the percentages of what we were looking for. This, however, is a bit hard to read. Let's try to display the largest percentage genre at the top and print in descending order. We will create another function to do so. 

We will make use of the `sorted()` function. The `sorted()`          function doesn't work too well with dictionaries because it only considers and returns the dictionary keys.

The `sorted()` function, however, works well if we transform the dictionary into a list of tuples, where each tuple contains a dictionary key along with its corresponding dictionary value. To ensure the sorting works right, the dictionary value comes first, and the dictionary key comes second.

In [20]:
def display_table_descending(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_value_as_tuple = (table[key], key)
        table_display.append(key_value_as_tuple)
        
    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])


Let's look at the frequency table for the `Category` section of the Google Play data set:

In [21]:
category_android = display_table_descending(android_free, 1)

FAMILY : 18.907942238267147
GAME : 9.724729241877256
TOOLS : 8.461191335740072
BUSINESS : 4.591606498194946
LIFESTYLE : 3.9034296028880866
PRODUCTIVITY : 3.892148014440433
FINANCE : 3.7003610108303246
MEDICAL : 3.531137184115524
SPORTS : 3.395758122743682
PERSONALIZATION : 3.3167870036101084
COMMUNICATION : 3.2378158844765346
HEALTH_AND_FITNESS : 3.0798736462093865
PHOTOGRAPHY : 2.944494584837545
NEWS_AND_MAGAZINES : 2.7978339350180503
SOCIAL : 2.6624548736462095
TRAVEL_AND_LOCAL : 2.33528880866426
SHOPPING : 2.2450361010830324
BOOKS_AND_REFERENCE : 2.1435018050541514
DATING : 1.861462093862816
VIDEO_PLAYERS : 1.7937725631768955
MAPS_AND_NAVIGATION : 1.3989169675090252
FOOD_AND_DRINK : 1.2409747292418771
EDUCATION : 1.1620036101083033
ENTERTAINMENT : 0.9589350180505415
LIBRARIES_AND_DEMO : 0.9363718411552346
AUTO_AND_VEHICLES : 0.9250902527075812
HOUSE_AND_HOME : 0.8235559566787004
WEATHER : 0.8009927797833934
EVENTS : 0.7107400722021661
PARENTING : 0.6543321299638989
ART_AND_DESIGN : 

We see that almost 19% are categorized as `FAMILY` and almost 10% are `GAME`. Let's see if this corresponds with the other column under `genre`.

In [22]:
genre_android = display_table_descending(android_free, 9)

Tools : 8.449909747292418
Entertainment : 6.069494584837545
Education : 5.347472924187725
Business : 4.591606498194946
Productivity : 3.892148014440433
Lifestyle : 3.892148014440433
Finance : 3.7003610108303246
Medical : 3.531137184115524
Sports : 3.463447653429603
Personalization : 3.3167870036101084
Communication : 3.2378158844765346
Action : 3.1024368231046933
Health & Fitness : 3.0798736462093865
Photography : 2.944494584837545
News & Magazines : 2.7978339350180503
Social : 2.6624548736462095
Travel & Local : 2.3240072202166067
Shopping : 2.2450361010830324
Books & Reference : 2.1435018050541514
Simulation : 2.0419675090252705
Dating : 1.861462093862816
Arcade : 1.8501805054151623
Video Players & Editors : 1.7712093862815883
Casual : 1.7599277978339352
Maps & Navigation : 1.3989169675090252
Food & Drink : 1.2409747292418771
Puzzle : 1.128158844765343
Racing : 0.9927797833935018
Role Playing : 0.9363718411552346
Libraries & Demo : 0.9363718411552346
Auto & Vehicles : 0.9250902527075

Here there is more specificity in the `genre` category. Tops genres include `Tools`, `Entertainment`, and `Education`, but those three together only account for about 20% of available genre labels. The difference between the `Category` and `Genre` labels are not readily clear. While we are most interested in the bigger picture, we will focus our analysis solely using the `Category` column.

Now let's compare Google Play's landscape to Apple's App Store:

In [23]:
genre_ios = display_table_descending(ios_free, 11)

Games : 58.16263190564867
Entertainment : 7.883302296710118
Photo & Video : 4.9658597144630665
Education : 3.662321539416512
Social Networking : 3.2898820608317814
Shopping : 2.60707635009311
Utilities : 2.5139664804469275
Sports : 2.1415270018621975
Music : 2.0484171322160147
Health & Fitness : 2.0173805090006205
Productivity : 1.7380509000620732
Lifestyle : 1.5828677839851024
News : 1.3345747982619491
Travel : 1.2414649286157666
Finance : 1.1173184357541899
Weather : 0.8690254500310366
Food & Drink : 0.8069522036002483
Reference : 0.5586592178770949
Business : 0.5276225946617008
Book : 0.4345127250155183
Navigation : 0.186219739292365
Medical : 0.186219739292365
Catalogs : 0.12414649286157665


Among the free English iOS apps, more than a half (58.16%) are games. Entertainment apps are close to 8%, followed by photo and video apps at close 5%. 

It seems that the App Store (the free English apps) is mostly contains apps that are for fun (games, entertainment, photo and video, social networking, sports, music, etc.), while apps with practical purposes (education, shopping, utilities, productivity, lifestyle, etc.) are less frequent. The Google Play data set seems to have a more balanced offerings of apps by category. We must keep in mind, however, that the most frequent categories may not mean it has the most users. Let's explore the areas where we have the most downloads.

## Kinds of Apps with the Most Users

One way to determine what genres are the most popular - i.e. have the most users - is to calculate the average number of installs of each app genre. For the Google Play data set, this can be found in the `Installs` column. The App Store, however, is missing this information. If we inspect the header row of the iOS data set, the next best thing we can analyze is the `rating_count_tot` or the total number of user ratings.

Let's begin by calculating the average number of user ratings per app genre on the App Store. To do that, we'll need to:

* Isolate the apps of each genre.
* Sum all the user ratings for the apps in each genre.
* Determine the total number of user ratings in each genre to calculate the average number of user ratings per app genre.

We will do this by creating a **nested loop**. The first loop will enter the genre. Then within each genre we will loop over the number of user ratings to do some calculations.

### iOS 

In [24]:
# Generates a frequency table for the prime_genre category column of the iOS
genre_ios = freq_table(ios_free, 11)

Now we will loop over the genres of the App Store data set. For each iteration or genre, we will initate a variable called `total` with a value of `0`, which will store the sum of the number of user ratings specific to each genre. We will iniate another variabe `len_genre` with a value of `0` to store the total number of apps specfic to each genre.

In [25]:
genre_ratings_ios = {}
for genre in genre_ios:
    total = 0
    len_genre = 0
    for row in ios_free:
        genre_app = row[11]
        if genre_app == genre:
            num_user_ratings = float(row[5])
            total += num_user_ratings
            len_genre += 1
    avg_num_ratings = total / len_genre
    genre_ratings_ios[genre] = avg_num_ratings
#    print(genre, ':', avg_num_ratings)

# print(genre_ratings_ios)

# To display in descending order
num_ratings_display = []
for key in genre_ratings_ios:
    key_value_as_tuple = (genre_ratings_ios[key], key)
    num_ratings_display.append(key_value_as_tuple)
        
table_sorted = sorted(num_ratings_display, reverse = True)
for entry in table_sorted:
    print(entry[1], ':', entry[0])

Navigation : 86090.33333333333
Reference : 74942.11111111111
Social Networking : 71548.34905660378
Music : 57326.530303030304
Weather : 52279.892857142855
Book : 39758.5
Food & Drink : 33333.92307692308
Finance : 31467.944444444445
Photo & Video : 28441.54375
Travel : 28243.8
Shopping : 26919.690476190477
Health & Fitness : 23298.015384615384
Sports : 23008.898550724636
Games : 22788.6696905016
News : 21248.023255813954
Productivity : 21028.410714285714
Utilities : 18684.456790123455
Lifestyle : 16485.764705882353
Entertainment : 14029.830708661417
Business : 7491.117647058823
Education : 7003.983050847458
Catalogs : 4004.0
Medical : 612.0


We see `Navigation` has the highest number of user ratings in the App Store.

But let's go deeper. Which `Navigation` apps are the most widely rated? Let's take a closer look...

In [26]:
for row in ios_free:
    if row[-5] == 'Navigation':
        print(row[1], ':', row[5]) # print name and number of ratings

Waze - GPS Navigation, Maps & Real-time Traffic : 345046
Google Maps - Navigation & Transit : 154911
Geocaching® : 12811
CoPilot GPS – Car Navigation & Offline Maps : 3582
ImmobilienScout24: Real Estate Search in Germany : 187
Railway Route Search : 5


Waze and Google Maps have the highest number of user ratings. Combined, they surpass half of all `Navigation` apps.

In [27]:
for row in ios_free:
    if row[-5] == 'Reference':
        print(row[1], ':', row[5]) # print name and number of ratings

Bible : 985920
Dictionary.com Dictionary & Thesaurus : 200047
Dictionary.com Dictionary & Thesaurus for iPad : 54175
Google Translate : 26786
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran : 18418
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition : 17588
Merriam-Webster Dictionary : 16849
Night Sky : 12122
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE) : 8535
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools : 4693
GUNS MODS for Minecraft PC Edition - Mods Tools : 1497
Guides for Pokémon GO - Pokemon GO News and Cheats : 826
WWDC : 762
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free : 718
VPN Express : 14
Real Bike Traffic Rider Virtual Reality Glasses : 8
教えて!goo : 0
Jishokun-Japanese English Dictionary & Translator : 0


We see that the Bible, Dictionary.com and the Google Translate app dominate this space.

In [28]:
for row in ios_free:
    if row[-5] == 'Social Networking':
        print(row[1], ':', row[5]) # print name and number of ratings

Facebook : 2974676
Pinterest : 1061624
Skype for iPhone : 373519
Messenger : 351466
Tumblr : 334293
WhatsApp Messenger : 287589
Kik : 260965
ooVoo – Free Video Call, Text and Voice : 177501
TextNow - Unlimited Text + Calls : 164963
Viber Messenger – Text & Call : 164249
Followers - Social Analytics For Instagram : 112778
MeetMe - Chat and Meet New People : 97072
We Heart It - Fashion, wallpapers, quotes, tattoos : 90414
InsTrack for Instagram - Analytics Plus More : 85535
Tango - Free Video Call, Voice and Chat : 75412
LinkedIn : 71856
Match™ - #1 Dating App. : 60659
Skype for iPad : 60163
POF - Best Dating App for Conversations : 52642
Timehop : 49510
Find My Family, Friends & iPhone - Life360 Locator : 43877
Whisper - Share, Express, Meet : 39819
Hangouts : 36404
LINE PLAY - Your Avatar World : 34677
WeChat : 34584
Badoo - Meet New People, Chat, Socialize. : 34428
Followers + for Instagram - Follower Analytics : 28633
GroupMe : 28260
Marco Polo Video Walkie Talkie : 27662
Miitomo : 2

We see the same domination we see in `Social Networking` as we do in `Navigation`. A select few dominate the space: Facebook, Pinterest, Skype, Messenger, Tumblr, and Whatsapp.

In [29]:
for row in ios_free:
    if row[-5] == 'Music':
        print(row[1], ':', row[5]) # print name and number of ratings

Pandora - Music & Radio : 1126879
Spotify Music : 878563
Shazam - Discover music, artists, videos & lyrics : 402925
iHeartRadio – Free Music & Radio Stations : 293228
SoundCloud - Music & Audio : 135744
Magic Piano by Smule : 131695
Smule Sing! : 119316
TuneIn Radio - MLB NBA Audiobooks Podcasts Music : 110420
Amazon Music : 106235
SoundHound Song Search & Music Player : 82602
Sonos Controller : 48905
Bandsintown Concerts : 30845
Karaoke - Sing Karaoke, Unlimited Songs! : 28606
My Mixtapez Music : 26286
Sing Karaoke Songs Unlimited with StarMaker : 26227
Ringtones for iPhone & Ringtone Maker : 25403
Musi - Unlimited Music For YouTube : 25193
AutoRap by Smule : 18202
Spinrilla - Mixtapes For Free : 15053
Napster - Top Music & Radio : 14268
edjing Mix:DJ turntable to remix and scratch music : 13580
Free Music - MP3 Streamer & Playlist Manager Pro : 13443
Free Piano app by Yokee : 13016
Google Play Music : 10118
Certified Mixtapes - Hip Hop Albums & Mixtapes : 9975
TIDAL : 7398
YouTube Mu

We also observe a coninued dominance in `Music` with `Pandora`, `Spotify` and `Shazam`.

As a general trend, the average number of ratings seem to be skewed by a select few apps in each genre, while other barely, if at all, pass the 10,000 user ratings mark. We could analyze this a step further and remove the extremely popular apps, but let's save this for later.

Based on the information that most of the apps offered in the App Store  are for fun (games, entertainment, photo and video, social networking, sports, music, etc.) combined with the new knowledge that `Navigation`, `Reference`, `Social Networking`, `Music`, and `Weather` are the top genres, I would target app development in the Reference Category primarily. There seems to be promise around the Bible - and combining it with 'Books' could be a winner. We could develop an app giving spiritual quotes every day, 'Bible for Children' app where main stories are shortened accompanied with illustrations, or perhaps a social network or people who would like to study the Bible together.

It seems attractive to put effort into utilitarian apps - such as weather and navigation since they have a majority of user ratings - but the average time spent on these apps would seem inconsequential for ad targeting due to the limited amount of time focused on these apps at a time. It may also require us to connect our apps to non-free live weather data APIs.

Social Networking apps also seem attractive, but unless the idea is novel, more convenient, and latches on quickly, monopolies such as Facebook and Pinterest will dominate the space. The barrier to entry is extremely high.

Let's dissect Google Play.

### Google Play
For the android data, we actually have the number of installs. We, however, do not have precise data on the number of installs - they are range. Let's assume that those with 1,000,000+ installs are 1,000,000 and those with 100,000 installs are 100,000. This way we can (albeit artificially) create a way to compate the app installs to each other. We don't have specifics on the number of installs, so this assumption is as good as we are going to get with the data that we have:

In [30]:
display_table_descending(android_free, 5) # Installs column

1,000,000+ : 15.726534296028879
100,000+ : 11.552346570397113
10,000,000+ : 10.548285198555957
10,000+ : 10.198555956678701
1,000+ : 8.393501805054152
100+ : 6.915613718411552
5,000,000+ : 6.825361010830325
500,000+ : 5.561823104693141
50,000+ : 4.7721119133574
5,000+ : 4.512635379061372
10+ : 3.5424187725631766
500+ : 3.2490974729241873
50,000,000+ : 2.3014440433213
100,000,000+ : 2.1322202166064983
50+ : 1.917870036101083
5+ : 0.78971119133574
1+ : 0.5076714801444043
500,000,000+ : 0.2707581227436823
1,000,000,000+ : 0.22563176895306858
0+ : 0.04512635379061372
0 : 0.01128158844765343


15.7% have over one million installs per each app, 11.55% have over 100,000+ and about 10% of apps have each over 10 million installs and 10,000. 

Let's fo further. But to do so, we'll have to convert each install number from a string to a float. This means we will need to remove the commas and the plus characters, otherwise the conversion will fail an raise an error.

To remove characters from strings, we use `str.replace(old, new)`. Just like `list.append()`, `str.replace(old, new)` is a kind of function called a method. `str.replace()` takes in two parameters, `old` and `new`, and replaces all occurences of `old` with the `new`.

In [31]:
category_android = freq_table(android_free, 1)

category_installs_android = {}

for category in category_android:
    total = 0
    len_category = 0
    for row in android_free:
        category_app = row[1]
        if category_app == category:            
            n_installs = row[5]
            n_installs = n_installs.replace(',', '')
            n_installs = n_installs.replace('+', '')
            total += float(n_installs)
            len_category += 1
    avg_installs = total / len_category
    category_installs_android[category] = avg_installs
    
# print(category_installs_android)
    
# To display in descending order
installs_display = []
for key in category_installs_android:
    key_value_as_tuple = (category_installs_android[key], key)
    installs_display.append(key_value_as_tuple)
        
table_sorted = sorted(installs_display, reverse = True)
for entry in table_sorted:
    print(entry[1], ':', entry[0])

COMMUNICATION : 38456119.167247385
VIDEO_PLAYERS : 24727872.452830188
SOCIAL : 23253652.127118643
PHOTOGRAPHY : 17840110.40229885
PRODUCTIVITY : 16787331.344927534
GAME : 15588015.603248259
TRAVEL_AND_LOCAL : 13984077.710144928
ENTERTAINMENT : 11640705.88235294
TOOLS : 10801391.298666667
NEWS_AND_MAGAZINES : 9549178.467741935
BOOKS_AND_REFERENCE : 8767811.894736841
SHOPPING : 7036877.311557789
PERSONALIZATION : 5201482.6122448975
WEATHER : 5074486.197183099
HEALTH_AND_FITNESS : 4188821.9853479853
MAPS_AND_NAVIGATION : 4056941.7741935486
FAMILY : 3695641.8198090694
SPORTS : 3638640.1428571427
ART_AND_DESIGN : 1986335.0877192982
FOOD_AND_DRINK : 1924897.7363636363
EDUCATION : 1833495.145631068
BUSINESS : 1712290.1474201474
LIFESTYLE : 1437816.2687861272
FINANCE : 1387692.475609756
HOUSE_AND_HOME : 1331540.5616438356
DATING : 854028.8303030303
COMICS : 817657.2727272727
AUTO_AND_VEHICLES : 647317.8170731707
LIBRARIES_AND_DEMO : 638503.734939759
PARENTING : 542603.6206896552
BEAUTY : 51315

In [32]:
for row in android_free:
    if row[1] == 'COMMUNICATION' and (row[5] == '1,000,000,000+'):
        print(row[0], ':', row[5])

WhatsApp Messenger : 1,000,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
Skype - free IM & video calls : 1,000,000,000+
Google Chrome: Fast & Secure : 1,000,000,000+
Gmail : 1,000,000,000+
Hangouts : 1,000,000,000+


In [33]:
for row in android_free:
    if row[1] == 'COMMUNICATION' and (row[5] == '500,000,000+'):
        print(row[0], ':', row[5])

Google Duo - High Quality Video Calls : 500,000,000+
imo free video calls and chat : 500,000,000+
LINE: Free Calls & Messages : 500,000,000+
UC Browser - Fast Download Private & Secure : 500,000,000+
Viber Messenger : 500,000,000+


In [34]:
for row in android_free:
    if row[1] == 'COMMUNICATION' and (row[5] == '100,000,000+'):
        print(row[0], ':', row[5])

imo beta free calls and text : 100,000,000+
Android Messages : 100,000,000+
Who : 100,000,000+
GO SMS Pro - Messenger, Free Themes, Emoji : 100,000,000+
Firefox Browser fast & private : 100,000,000+
Messenger Lite: Free Calls & Messages : 100,000,000+
Kik : 100,000,000+
KakaoTalk: Free Calls & Text : 100,000,000+
Opera Mini - fast web browser : 100,000,000+
Opera Browser: Fast and Secure : 100,000,000+
Telegram : 100,000,000+
Truecaller: Caller ID, SMS spam blocking & Dialer : 100,000,000+
UC Browser Mini -Tiny Fast Private & Secure : 100,000,000+
WeChat : 100,000,000+
Yahoo Mail – Stay Organized : 100,000,000+
BBM - Free Calls & Messages : 100,000,000+


On average, communication apps have the most installs at 38,456,119. A few apps doinate this space with over one billion installs (WhatsApp, Facebook Messenger, Skype, Google Chrome, Gmail, and Hangouts), and a few others with over 100 and 500 million installs.

Let's try to grasp the monopoly on the apps that have over 100 million installs. If we remove them, the average number of installs goes down dramatically:

In [35]:
under_100_mil = []

for row in android_free:
    n_installs = row[5]
    n_installs = n_installs.replace(',', '')
    n_installs = n_installs.replace('+', '')
    if (row[1] == 'COMMUNICATION') and (float(n_installs) < 100000000):
        under_100_mil.append(float(n_installs))
        
sum(under_100_mil) / len(under_100_mil)

3603485.3884615386

We see that removing the `COMMUNICATION` apps with over 100 million installs, brings the average number of installs to just above 3.5 million (about a 10 fold decrease - from 38,456,119).

Let's look at the other top categories including `VIDEO_PLAYERS`, `SOCIAL`, and `PHOTOGRAPHY`:

In [36]:
for row in android_free:
    if row[1] == 'VIDEO_PLAYERS' and (row[5] == '1,000,000,000+'):
        print(row[0], ':', row[5])

YouTube : 1,000,000,000+
Google Play Movies & TV : 1,000,000,000+


In [37]:
for row in android_free:
    if row[1] == 'VIDEO_PLAYERS' and (row[5] == '500,000,000+'):
        print(row[0], ':', row[5])

MX Player : 500,000,000+


In [38]:
for row in android_free:
    if row[1] == 'VIDEO_PLAYERS' and (row[5] == '100,000,000+'):
        print(row[0], ':', row[5])

Motorola Gallery : 100,000,000+
VLC for Android : 100,000,000+
Dubsmash : 100,000,000+
VivaVideo - Video Editor & Photo Movie : 100,000,000+
VideoShow-Video Editor, Video Maker, Beauty Camera : 100,000,000+
Motorola FM Radio : 100,000,000+


The most popular are YouTube and Google Play Movies & TV with over 1 billion installs. MX Player is third with over 500 million and a handfull qualify for over 100 million installs.

In [39]:
for row in android_free:
    if row[1] == 'SOCIAL' and (row[5] == '1,000,000,000+'):
        print(row[0], ':', row[5])

Facebook : 1,000,000,000+
Google+ : 1,000,000,000+
Instagram : 1,000,000,000+


In [40]:
for row in android_free:
    if row[1] == 'SOCIAL' and (row[5] == '500,000,000+'):
        print(row[0], ':', row[5])

Facebook Lite : 500,000,000+
Snapchat : 500,000,000+


In [41]:
for row in android_free:
    if row[1] == 'SOCIAL' and (row[5] == '100,000,000+'):
        print(row[0], ':', row[5])

Tumblr : 100,000,000+
Pinterest : 100,000,000+
Badoo - Free Chat & Dating App : 100,000,000+
Tango - Live Video Broadcast : 100,000,000+
LinkedIn : 100,000,000+
Tik Tok - including musical.ly : 100,000,000+
BIGO LIVE - Live Stream : 100,000,000+
VK : 100,000,000+


Facebook, Google+, and Instagram dominate the space with over 1 billion downloads. Facebook Lite and Snapchat have over 500 million and a handful, including Tumblr, Pinterest, LinkedIn, and Tik Tok, have over 100 million installs.

In [42]:
for row in android_free:
    if row[1] == 'PHOTOGRAPHY' and (row[5] == '1,000,000,000+'):
        print(row[0], ':', row[5])

Google Photos : 1,000,000,000+


In [43]:
for row in android_free:
    if row[1] == 'PHOTOGRAPHY' and (row[5] == '500,000,000+'):
        print(row[0], ':', row[5])

In [44]:
for row in android_free:
    if row[1] == 'PHOTOGRAPHY' and (row[5] == '100,000,000+'):
        print(row[0], ':', row[5])

B612 - Beauty & Filter Camera : 100,000,000+
YouCam Makeup - Magic Selfie Makeovers : 100,000,000+
Sweet Selfie - selfie camera, beauty cam, photo edit : 100,000,000+
Retrica : 100,000,000+
Photo Editor Pro : 100,000,000+
BeautyPlus - Easy Photo Editor & Selfie Camera : 100,000,000+
PicsArt Photo Studio: Collage Maker & Pic Editor : 100,000,000+
Photo Collage Editor : 100,000,000+
Z Camera - Photo Editor, Beauty Selfie, Collage : 100,000,000+
PhotoGrid: Video & Pic Collage Maker, Photo Editor : 100,000,000+
Candy Camera - selfie, beauty camera, photo editor : 100,000,000+
YouCam Perfect - Selfie Photo Editor : 100,000,000+
Camera360: Selfie Photo Editor with Funny Sticker : 100,000,000+
S Photo Editor - Collage Maker , Photo Collage : 100,000,000+
AR effect : 100,000,000+
Cymera Camera- Photo Editor, Filter,Collage,Layout : 100,000,000+
LINE Camera - Photo editor : 100,000,000+
Photo Editor Collage Maker Pro : 100,000,000+


Again, we see domination with Google Photos. We do not, however, see any over 500 million and a substantial handful over 100 million. This could be a sweet spot for the company to develop a photography app. It could be an editor or personal photo journal. This space is pretty saturated, so it might make more sense to look at categories that have less installs.

We need to keep in mind that more installs does not necessarily mean that a lot of people use it and/or spend a lot of time on the app.

Let's look at categories that by its nature, require a little more time on the app. We have already discussed photography - journals or editors. This type of app would require some minutes to be on. This might be another advantage to developing a photography app - more minutes spent on the app will mean more opportunties for the user to see an ad.

But let's look at other categories such as `PRODUCTIVITY`, `NEWS_AND_MAGAZINES` and `BOOKS_AND_REFERENCE`.

In [45]:
for row in android_free:
    if row[1] == 'PRODUCTIVITY' and (row[5] == '1,000,000,000+'
                             or row[5] == '500,000,000+'
                             or row[5] == '100,000,000+'):
        print(row[0], ':', row[5])

Microsoft Word : 500,000,000+
Microsoft Outlook : 100,000,000+
Microsoft OneDrive : 100,000,000+
Microsoft OneNote : 100,000,000+
Google Keep : 100,000,000+
ES File Explorer File Manager : 100,000,000+
Dropbox : 500,000,000+
Google Docs : 100,000,000+
Microsoft PowerPoint : 100,000,000+
Samsung Notes : 100,000,000+
SwiftKey Keyboard : 100,000,000+
Google Drive : 1,000,000,000+
Adobe Acrobat Reader : 100,000,000+
Google Sheets : 100,000,000+
Microsoft Excel : 100,000,000+
WPS Office - Word, Docs, PDF, Note, Slide & Sheet : 100,000,000+
Google Slides : 100,000,000+
ColorNote Notepad Notes : 100,000,000+
Evernote – Organizer, Planner for Notes & Memos : 100,000,000+
Google Calendar : 500,000,000+
Cloud Print : 500,000,000+
CamScanner - Phone PDF Creator : 100,000,000+


These productivity apps seem like they all have a clear purpose where users would be 'in' and then 'out' of the app very quickly. Especially on a mobile app, it would seem that very few people would use documentation editing tools. This space, however, could be ripe for a clean, easy-to-use, documentation viewer and/or editor to make collaborative changes. By the nature of these apps, users will be spending time in the app, which gives us more opportunity to use ad targeting.

This could be a novel space, albeit risky, to develop a colloborative mobile document editor.

Let's look at perhaps less-risk areas such as `NEWS_AND_MAGAZINES` as well as `BOOKS_AND_REFERENCE`. Keep in mind, an app surrounding a book or books in the iOS App Store, was suggested. Having a similar suggestion for Google Play, could cast our net wider without creating two completely separate apps to uniquely cater to both Google Play and the App Store.

In [46]:
for row in android_free:
    if row[1] == 'NEWS_AND_MAGAZINES' and (row[5] == '1,000,000,000+'
                             or row[5] == '500,000,000+'
                             or row[5] == '100,000,000+'):
        print(row[0], ':', row[5])

Twitter : 500,000,000+
Flipboard: News For Our Time : 500,000,000+
Google News : 1,000,000,000+


In [47]:
for row in android_free:
    if row[1] == 'BOOKS_AND_REFERENCE' and (row[5] == '1,000,000,000+'
                             or row[5] == '500,000,000+'
                             or row[5] == '100,000,000+'):
        print(row[0], ':', row[5])

Google Play Books : 1,000,000,000+
Bible : 100,000,000+
Amazon Kindle : 100,000,000+
Wattpad 📖 Free Books : 100,000,000+
Audiobooks from Audible : 100,000,000+


Both of these areas have very few popular apps, so it could be worth explorings these categories to develop other apps.

Let's tlook at the kinds of apps that are in the middle in terms of popularity (between 1,000,000 and 100,000,000 downloads):

In [48]:
for row in android_free:
    if row[1] == 'NEWS_AND_MAGAZINES' and (row[5] == '1,000,000+'
                             or row[5] == '5,000,000+'
                             or row[5] == '10,000,000+'
                             or row[5] == '50,000,000+'):
        print(row[0], ':', row[5])

print('\n')
        
for row in android_free:
    if row[1] == 'BOOKS_AND_REFERENCE' and (row[5] == '1,000,000+'
                             or row[5] == '5,000,000+'
                             or row[5] == '10,000,000+'
                             or row[5] == '50,000,000+'):
        print(row[0], ':', row[5])

Fox News – Breaking News, Live Video & News Alerts : 10,000,000+
NEW - Read Newspaper, News 24h : 10,000,000+
BaBe + - Indonesian News : 1,000,000+
daily News : 1,000,000+
BBC News : 10,000,000+
Free TV Shows App:News, TV Series, Episode, Movies : 1,000,000+
News24 : 1,000,000+
Le Monde, the continuous news : 5,000,000+
Wireless news : 1,000,000+
CNN Breaking US & World News : 10,000,000+
Gnoche entertainment news · sports news is also free : 5,000,000+
BaBe - Read News : 10,000,000+
Nigeria News NAIJ.com : 1,000,000+
detikcom - Latest & Most Complete News : 10,000,000+
Dailyhunt (Newshunt) - Latest News, Viral Videos : 50,000,000+
BaBe Lite - Read Quota Saving News : 1,000,000+
Bengali Newspaper - The first L. : 1,000,000+
Read- Latest News, Information, Gossip and Politics : 10,000,000+
Reddit: Social News, Trending Memes & Funny Videos : 10,000,000+
Opera News - Trending news and videos : 10,000,000+
Topbuzz: Breaking News, Videos & Funny GIFs : 10,000,000+
Pulse Nabd - World News, 

The news seems to be well diversified and saturated so let's look at `BOOKS_AND_REFERENCE`. It seems it is dominated by software for processing and reading ebooks, dictionaries, and librariers. We will avoid these areas since there is significant competition.

We made the suggestion of centering our app around a book with the App Store. This could be an attractive path here in Google Play as well. We see that there are multiple apps surrounding the Quran suggesting that we could develop an app based off of a more recent book with built-in dictionaries to consolidate and streamline the user experience. We could do audiobooks, quotes from books, a social forum, or virtual book-club to name a few.


## Conclusions
We analyzed data from the App Store and Google Play mobile apps with the aim of identifying and recommending an app profile that can be profitable for both markets by using ad-targeting.

We would like apps that a lot of people install, are a popular genre, and will spend some time on.

We concluded that an app surroudning a book could be profitable for both the Google Play and the App Store markets. This could be audiobooks, quotes from books, a social forum, a book-review forum, what famous people are reading i.e. spin off on Gates' suggested reads, or virtual book-club to name a few.